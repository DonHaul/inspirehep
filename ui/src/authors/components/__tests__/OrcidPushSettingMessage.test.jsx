import React from 'react';
import { render } from '@testing-library/react';

import OrcidPushSettingMessage from '../OrcidPushSettingMessage';

describe('OrcidPushSettingMessage', () => {
  it('renders when enabled', () => {
    const { asFragment } = render(
      <OrcidPushSettingMessage orcid="0000-0001-8058-0014" enabled />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when disabled', () => {
    const { asFragment } = render(
      <OrcidPushSettingMessage orcid="0000-0001-8058-0014" enabled={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
