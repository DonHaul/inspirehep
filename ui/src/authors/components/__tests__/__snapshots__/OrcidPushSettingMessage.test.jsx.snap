// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrcidPushSettingMessage renders when disabled 1`] = `
<DocumentFragment>
  <div
    data-testid="orcid-push-setting-message"
  >
    <p>
      Your INSPIRE works are not exported to your ORCID yet. Please note that only the publications that are verified as yours on INSPIRE will be exported to ORCID.
    </p>
  </div>
</DocumentFragment>
`;

exports[`OrcidPushSettingMessage renders when enabled 1`] = `
<DocumentFragment>
  <div
    data-testid="orcid-push-setting-message"
  >
    <p>
      This profile is already connected to the following ORCID: 
      <a
        href="//orcid.org/0000-0001-8058-0014"
        target="_blank"
      >
        0000-0001-8058-0014
      </a>
    </p>
    <p>
      Your claimed works will be exported automatically.
    </p>
  </div>
</DocumentFragment>
`;
